### PR TITLE
obfuscated rar sets: better handling missing rars

### PR DESCRIPTION
**Pull request by sanderjo**
As described in https://github.com/sabnzbd/sabnzbd/issues/1687

And added two unit test for these cases.

### CODE REVIEW

**Comment by sanderjo** [comment]
Closes #1687 , Black and CI are happy, so ... @Safihre ... are you willing to accept?

File: sabnzbd/postproc.py [hunk 1]
```diff
@@ -948,47 +952,77 @@ def rar_renamer(nzo: NzbObject, workdir):
...
+
+    # Sanity check of the rar set
+    highest_rar = sorted(rarvolnr.keys())[-1]
+    """
+    # First an easy check
+    if highest_rar != len(rarvolnr):
+        # there is a missing rar file, so
+        logging.warning("Incomplete rar set, so I can't deobfuscate")
+        return 0
+    """
```
Comment (Safihre) [inline sabnzbd/postproc.py:967]: Please remove unused code.

File: sabnzbd/postproc.py [hunk 2]
```diff
@@ -948,47 +952,77 @@ def rar_renamer(nzo: NzbObject, workdir):
...
+    # More intelligent check aka staircase check: number of rarsets should no go up, but stay the same or go down
+    how_many_previous = 1000  # 1000 rarset mixed ... should be enough ...
+    for rar_set_number in range(1, highest_rar + 1):
+        try:
+            how_many_here = len(rarvolnr[rar_set_number])
+        except:
+            # rarset does not exist at all
+            logging.warning("rarset %s is missing completely, so I can't deobfuscate.", rar_set_number)
```
Comment (Safihre) [inline sabnzbd/postproc.py:975]: So this works well if one rar-set has 10 files and the other 15? Almost looks like it requires all sets to have same amount of rar's?

Comment (sanderjo) [reply sabnzbd/postproc.py:975]: Yes, that works. Please note that this was existing code (with just indentation changed)

File: sabnzbd/postproc.py (hunk 1 above)
Comment (sanderjo) [reply sabnzbd/postproc.py:967]: OK, done

**Pull request merged**